docs/flink-getting-started.md

```diff
@@ -683,7 +683,47 @@ env.execute("Test Iceberg DataStream");
+| snapshot-id                 |                                    | For time travel in batch mode. Read data from the specified snapshot-id. |
+| case-sensitive              | false                              | Whether the sql is case sensitive                            |
+| as-of-timestamp             |                                    | For time travel in batch mode. Read data from the most recent snapshot as of the given time in milliseconds. |
+| starting-strategy           | INCREMENTAL_FROM_LATEST_SNAPSHOT   | Starting strategy for streaming execution. TABLE_SCAN_THEN_INCREMENTAL: Do a regular table scan then switch to the incremental mode. The incremental mode starts from the current snapshot exclusive. INCREMENTAL_FROM_LATEST_SNAPSHOT: Start incremental mode from the latest snapshot inclusive. If it is an empty map, all future append snapshots should be discovered. INCREMENTAL_FROM_EARLIEST_SNAPSHOT: Start incremental mode from the earliest snapshot inclusive. If it is an empty map, all future append snapshots should be discovered. INCREMENTAL_FROM_SNAPSHOT_ID: Start incremental mode from a snapshot with a specific id inclusive. INCREMENTAL_FROM_SNAPSHOT_TIMESTAMP: Start incremental mode from a snapshot with a specific timestamp inclusive. If the timestamp is between two snapshots, it should start from the snapshot after the timestamp. Just for FIP27 Source |
+| start-snapshot-timestamp    |                                    | Start to read data from the most recent snapshot as of the given time in milliseconds. |
+| start-snapshot-id           |                                    | Start to read data from the specified snapshot-id.           |
+| end-snapshot-id             | The latest snapshot id             | Specifies the end snapshot.                                  |
+| split-size                  | Table read.split.target-size       | Overrides this table's read.split.target-size                |
+| split-lookback              | Table read.split.planning-lookback | Overrides this table's read.split.planning-lookback          |
+| split-file-open-cost        | Table read.split.open-file-cost    | Overrides this table's read.split.open-file-cost             |
+| streaming                   | false                              | Sets whether the current task runs in streaming or batch mode. |
+| monitor-interval            | 10s                                | Interval for listening on the generation of new snapshots.   |
+| include-column-stats        | false                              | Create a new scan from this that loads the column stats with each data file. Column stats include: value count, null value count, lower bounds, and upper bounds. |
+| max-planning-snapshot-count | Integer.MAX_VALUE                  | If there are multiple new snapshots, configure the maximum number of snapshot forward at a time. |
```

Comment:
How about changing the default value to 1? In another PR. @stevenzwu

Reply:
@chenjunjiedada I am in favor of changing this config default. this is only applicable for streaming job. but I would probably set the default to some number higher like 6 or 10. This is actually very relevant to PR #6299  that I created recently.

Reply:
+1 to change to 10.

Reply:
I have some conflicts here. If we make it smaller, will it take a long time to catch up to the latest snapshot  when the data backlog is severe (such as after a restart)?

Reply:
Unless there are mass small files making the bottleneck polling snapshot and reading manifest, otherwise I think the bottleneck would be reading the data file.

Reply:
typically the split discovery interval should be <= upstream commit interval. when the job is provisioned properly for steady state, it shouldn't be able to process 10 snapshots/commits worth of data in one discovery cycle. that is like 10x of steady-state throughput, which seems like big over-provision. that is my intuition for default like 10.

if the job is scaled up 10x for faster catch-up, that is where we can potentially have issue with default at 10. is it reasonable to ask users to change the `max-planning-snapshot-count` config in this case?